Utilities/DavixAdaptor/plugins/DavixStorageMaker.cc

```diff
@@ -0,0 +1,50 @@
+#include "FWCore/Utilities/interface/Exception.h"
+#include "Utilities/DavixAdaptor/interface/DavixFile.h"
+#include "Utilities/StorageFactory/interface/StorageFactory.h"
+#include "Utilities/StorageFactory/interface/StorageMaker.h"
+#include "Utilities/StorageFactory/interface/StorageMakerFactory.h"
+#include <davix.hpp>
+#include <unistd.h>
+
+class DavixStorageMaker : public StorageMaker {
+
+public:
+  /** Open a storage object for the given URL (protocol + path), using the
+      @a mode bits.  No temporary files are downloaded.  */
+  virtual std::unique_ptr<Storage> open(const std::string &proto, const std::string &path, int mode,
+                                        AuxSettings const &aux) const override {
+    const StorageFactory *f = StorageFactory::get();
+    std::string newurl((proto == "web" ? "http" : proto) + ":" + path);
+    auto file = std::make_unique<DavixFile>(newurl, mode);
+    return f->wrapNonLocalFile(std::move(file), proto, std::string(), mode);
+  }
+
+  virtual bool check(const std::string &proto, const std::string &path, const AuxSettings &aux,
+                     IOOffset *size = 0) const override {
+    std::string newurl((proto == "web" ? "http" : proto) + ":" + path);
+    Davix::DavixError *err = NULL;
+    std::unique_ptr<Davix::DavPosix> davixPosix =
```

Comment:
There doesn't appear to be a reason to use a `std::unique_ptr<Davix::DavPosix>` here. Instead just create one on the stack
```C++
Davix::DavPosix davixPosix;
```